### PR TITLE
Ribbon gls example

### PR DESCRIPTION
**Pull request by AmishgaAlphonius**
A new "Ribbon Mixer using a Single Rotating Reference Frame" example was created.

# Documentation

- Change "Compute Canada cluster" for "Digital Research Alliance of Canada cluster" in the installation guide

- [x] doc/source/installation/compute_canada.rst

- Add _Velocity Source_ page in the _Parameters Guide_

- [x] doc/source/parameters/cfd/cfd.rst
- [x] doc/source/parameters/cfd/velocity_source.rst

# Comments

- Should we add a "Velocity Source" page in the _Parameter Guide_ for the "Velocity Source" subsection used in this example? -> It's added now!

### CODE REVIEW

File: doc/source/examples/incompressible-flow/3d_mixer_using_single_rotating_frame/3d_mixer_using_single_rotating_frame.rst [hunk 1]
```diff
@@ -0,0 +1,496 @@
...
+Simulating for a Specific Flow Condition :math:`(Re = 1)`
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+In the ``output`` subdirectory, you will find the ``mixer.pvd`` file that can be visualized using `Paraview <https://www.paraview.org/download/>`_. In the figure below, the velocity magnitude are shown for a flow at :math:`Re = 1`. Because a SRF is used, we can notice the rotational velocity imposed on the walls and the no-slip condition on the ribbon.
+
+.. image:: images/velocity_magnitude.png
+   :alt: Ribbon Mixer Velocity magnitude
+   :align: center
+   :name: Ribbon Mixer Velocity magnitude
+
+In reality, the velocity field and its streamlines are visualized in an Eulerian frame. It is possible to do so by simply selecting ``velocity_eulerian`` in Paraview.
```
Comment (CaroleAnneDaunais) [inline doc/source/examples/incompressible-flow/3d_mixer_using_single_rotating_frame/3d_mixer_using_single_rotating_frame.rst:433]:
```suggestion
In the reference frame of the tank, the velocity field and its streamlines are visualized in an Eulerian frame. It is possible to do so by simply selecting ``velocity_eulerian`` in Paraview.
```

**Review by CaroleAnneDaunais — COMMENTED** [review]
Nice example! The Compute Canada manipulations are cleverly reported. This is going to be very helpful for everyone! :)

**Comment by blaisb** [comment]
@AmishgaAlphonius I would add the velocity source to the RST
Will gladly review PR tomorrow

**Comment by blaisb** [comment]
I also suggest you rebase, will make the CI simpler

File: doc/source/examples/incompressible-flow/3d_mixer_using_single_rotating_frame/3d_mixer_using_single_rotating_frame.rst [hunk 2]
```diff
@@ -0,0 +1,496 @@
...
+   :align: center
+   :name: Power curve
+
+As you can see, :math:`N_p` decreases when :math:`Re` increases. It is explained by:
+
+.. math::
+
+    N_p = \frac{2 \pi \Gamma}{\rho N^2 D^5} =\frac{1}{Re}  \frac{2 \pi \Gamma}{\mu  N D^3}
+
+As we can see in the image above, for :math:`Re < 10` (laminar regime), the Lethe curve fits well with the experimental values.
```
Comment (blaisb) [inline doc/source/examples/incompressible-flow/3d_mixer_using_single_rotating_frame/3d_mixer_using_single_rotating_frame.rst:492]:
```suggestion
As we can see in the image above, for :math:`Re < 100` (laminar regime), the Lethe curve fits well with the experimental values.
```

File: doc/source/examples/incompressible-flow/3d_mixer_using_single_rotating_frame/3d_mixer_using_single_rotating_frame.rst [hunk 3]
```diff
@@ -0,0 +1,496 @@
...
+
+References
```
Comment (blaisb) [inline doc/source/examples/incompressible-flow/3d_mixer_using_single_rotating_frame/3d_mixer_using_single_rotating_frame.rst:494]: I would add a possibility for extension to suggest simulating it at a higher Reynolds number in a transient simulation and compare with the experimental data :).

**Review by blaisb — CHANGES_REQUESTED** [review]
Very minor comments. Good @AmishgaAlphonius !!!!

File: doc/source/examples/incompressible-flow/3d_mixer_using_single_rotating_frame/3d_mixer_using_single_rotating_frame.rst [hunk 4]
```diff
@@ -0,0 +1,496 @@
...
+When designing an industrial mixer, it is essential to evaluate the power consumed by the agitator. To do so, we usually refer to graphs correlating the power number (:math:`N_p`) as a function of the Reynolds number (:math:`Re`). :math:`N_p` is a dimensionless number that relates the power consumed by the agitator to the geometry of the system, the speed of the agitator, and the properties of the material being mixed (density :math:`\rho` and viscosity :math:`\mu`). In this example, using a single rotating frame (SRF) model, we simulate a fluid being mixed in a ribbon mixer for different flow conditions (different values of :math:`Re`) to generate :math:`N_p` vs :math:`Re` curves.
+
+
+Features
+-------------
+- Solver: ``gls_navier_stokes_3d`` (with Q1-Q1)
+- Steady-state problem
+- Displays the use of a single rotating frame (``srf``) when modeling a complex rotating geometry
+- Calculation of the power number (:math:`N_p`) using a python script
+- Use of a Digital Research Alliance of Canada (Alliance) cluster to run simulations
```
[the remark this answers is not in the document]
Comment (AmishgaAlphonius) [reply doc/source/examples/incompressible-flow/3d_mixer_using_single_rotating_frame/3d_mixer_using_single_rotating_frame.rst:14]: I think the idea behind it was to let readers know that this is a first example where we go through step-by-step on how to launch simulations on a cluster and then retrieve the results from it. I might be interpreting the "Features" section wrong.

File: doc/source/examples/incompressible-flow/3d_mixer_using_single_rotating_frame/3d_mixer_using_single_rotating_frame.rst [hunk 5]
```diff
@@ -0,0 +1,496 @@
...
+where
+
+- :math:`N` is the angular velocity :math:`[\text{rad} \cdot \text{s}^{-1}]` of the agitator;
+- :math:`D` is the diameter of the impeller [m], and
+- :math:`\nu` is the kinematic viscosity of the fluid :math:`[\text{m}^2 \cdot \text{s}^{-1}]`.
+
+In order to generate :math:`N_p` vs :math:`Re` curves, we have to calculate :math:`N_p`. In this example, we will be using the torque (:math:`\Gamma`) to calculate :math:`N_p` with the following expression:
+
+.. math::
+    N_p = \frac{2 \pi \Gamma}{\rho N^2 D^5}
```
[the remark this answers is not in the document]
Comment (AmishgaAlphonius) [reply doc/source/examples/incompressible-flow/3d_mixer_using_single_rotating_frame/3d_mixer_using_single_rotating_frame.rst]: I've moved it up :)

File: doc/source/examples/incompressible-flow/3d_mixer_using_single_rotating_frame/3d_mixer_using_single_rotating_frame.rst [hunk 6]
```diff
@@ -0,0 +1,496 @@
...
+.. seealso::
+
+    If it is your first time running simulations from Lethe on an Alliance cluster, you may want to see our installation guide on how to setup and install all the necessary software and modules: :doc:`../../../installation/compute_canada`.
+
+.. _Generating_the_Different_Cases:
+
+Generating the Different Cases
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Using ``lethe_case_generator.py``, we generate the :math:`25` cases with :math:`Re` ranging from :math:`0.1` to :math:`100`. Before running the Python script, it is important to **specify your group account name**, next to ``#SBATCH --account=`` among the job directives of the ``launch_mixer.sh`` script located in the ``template`` folder.
```
[the remark this answers is not in the document]
Comment (AmishgaAlphonius) [reply doc/source/examples/incompressible-flow/3d_mixer_using_single_rotating_frame/3d_mixer_using_single_rotating_frame.rst:283]: I've added a few sentences, can you see if it's okay?

File: doc/source/examples/incompressible-flow/3d_mixer_using_single_rotating_frame/3d_mixer_using_single_rotating_frame.rst [hunk 7]
```diff
@@ -0,0 +1,496 @@
...
+
+.. warning::
+
+	In order to run a job on an Alliance cluster, it is required to at least specify the time limit of the job (``-- time``) and your account (``--account``).
+
+
+Here are a few examples of other job directives you may want to specify in your job script:
```
[the remark this answers is not in the document]
Comment (AmishgaAlphonius) [reply doc/source/examples/incompressible-flow/3d_mixer_using_single_rotating_frame/3d_mixer_using_single_rotating_frame.rst:289]: I've added a "warning", can you see if it's okay?

File: doc/source/examples/incompressible-flow/3d_mixer_using_single_rotating_frame/3d_mixer_using_single_rotating_frame.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (blaisb) [reply doc/source/examples/incompressible-flow/3d_mixer_using_single_rotating_frame/3d_mixer_using_single_rotating_frame.rst:26]: Nah your interpretation is correct. I did not see it that way initially, but it's a good idea :)

**Review by ghazalemir — COMMENTED** [review]
Amishga,
I went through the example and also Carole-Anne and Bruno's comments; I had very minimal comments more than what they commented on
Good job for explaining all necessary details.

**Comment by blaisb** [comment]
@AmishgaAlphonius I added the Navier-Stokes equation in a rotating frame of reference :).

**Review by CaroleAnneDaunais — APPROVED** [review]

File: doc/source/parameters/cfd/velocity_source.rst [hunk 8]
```diff
@@ -0,0 +1,23 @@
+Velocity Source
+~~~~~~~~~~~~~~~~
+
+This subsection allows you to change the reference frame of the calculated velocity in a given system. This is especially useful when it is easier to solve a problem in a Lagrangian reference frame.
```
Comment (blaisb) [inline doc/source/parameters/cfd/velocity_source.rst:4]:
```suggestion
This subsection allows you to defined velocity-dependant source term. At the moment, this used to solve a problems in a Lagrangian reference frame for which the Coriolis and centrifugal forces must be added.
```

**Review by blaisb — APPROVED** [review]

**Pull request merged**